src/test/java/com/nimbusds/jose/JWEObjectJSONTest.java

```diff
@@ -400,16 +404,411 @@ public void test_ecdh_1pu_a128kw_appendix() throws Exception {
                 "      \"Az2IWsISEMDJvyc5XRL-3-d-RgNBOGolCsxFFoUXFYw\"," +
                 "     \"tag\":" +
                 "      \"HLb4fTlm8spGmij3RyOs2gJ4DpHM4hhVRwdF_hGb3WQ\"" +
-                "    }");
+                "    }";
 
+        // both key decryption
+        JWEObjectJSON jwe = JWEObjectJSON.parse(json);
         List<Pair<UnprotectedHeader, OctetKeyPair>> recipients = Arrays.asList(
                 Pair.of(new UnprotectedHeader.Builder().keyID("bob-key-2").build(), bobKey),
                 Pair.of(new UnprotectedHeader.Builder().keyID("2021-05-06").build(), charlieKey)
         );
 
         ECDH1PUX25519DecrypterMulti decrypterMulti = new ECDH1PUX25519DecrypterMulti(aliceKey.toPublicJWK(), recipients);
         jwe.decrypt(decrypterMulti);
+        assertEquals("Three is a magic number.", jwe.getPayload().toString());
 
+        // bob key decryption
+        jwe = JWEObjectJSON.parse(json);
+        recipients = Collections.singletonList(Pair.of(new UnprotectedHeader.Builder().keyID("bob-key-2").build(), bobKey));
```

Comment:
Please add a test where passed KIDs are not just magic numbers, but get from the parsed JSON (as real application will do to find out the KIDs)